backend/src/main/kotlin/com/tngtech/apicenter/backend/config/RestResponseExceptionHandler.kt

```diff
@@ -0,0 +1,48 @@
+package com.tngtech.apicenter.backend.config
+
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.springframework.web.bind.annotation.ExceptionHandler
+import java.time.Instant
+import java.time.format.DateTimeFormatter
+
+@ControllerAdvice
+class RestResponseExceptionHandler() {
+
+    @ExceptionHandler(SpecificationNotFoundException::class)
+    fun handleNotFound(): ResponseEntity<ErrorMessage> {
+        return ResponseEntity(
```

Comment:
What do you think about returning an actual object that could be serialized to JSON, instead of only the message as a string? Often, you can then add both a machine-readable error code (f.e. `SPECIFICATION_NOT_FOUND`) and a human-readable message with more details ("could not find the specification $name").
Additionally, I've seen the concept of logrefs: A UUID that is passed to the user and noted in the log so that debugging the problem gets easier.

Reply:
This is already the case, it's just the second field is set automatically, ie. the client receives `{"message":"Specification needs a title","timestamp":"2019-02-27T14:25:05.508Z"}`. I just put the bare minumum fields in this commit.

Reply:
What do you think about the machine-readable error code and a separate message?

Reply:
I've nothing against it, but I'm more motivated by only what fields the application needs in it's current state. Is there a convention / spec for API responses you could point me to, if these fields are expected?

Reply:
No, there is no convetion. However, I've learned the need for both fields in previous projects, as a machine readable code cannot contain details, but you need them for actual debugging. So, for me at least, it's a good practice that I'd like to spread.